server/controllers/userController.js

```diff
@@ -0,0 +1,29 @@
+import { Router } from "express";
+import { userService } from "../services";
+
+// This should be refactored, and sessions should be stored in mongo!
+const sessions = [];
```

Comment:
This won't actually work, because it will be out of sync the with session in your index file. I just added it here so the code I copied over would compile. 

Having session like this is fine for local use, but once we start hosting the app it will cause some issues. Heroku, and most other hosting platforms, use _containers_ to run your application (in herokus case, they call them _dynos_). This makes the application ephemeral, in that if no one is using it heroku will spin the app down to reclaim resources. Since session will only be stored in RAM, this means that they'll die with the app instance on the dyno.

A better solution here would be to add an _authentication middleware_ and check against a database (or some other more permanent storage, for this we probably want to use a database called Redis).

I can help you plan it after we finish the migration!